### ISSUE TITLE
Non-public class methods get overlooked when checking `missing-docstring`

### ISSUE THREAD

**Issue by joshuahaertel** [PyCQA/pylint#1866]
### Steps to reproduce
1. Create a class with a `_` or `__` prefixed method that _doesn't_ have a docstring.
2. For clarity, make sure the class and the module have a docstring with some text in it.
3. Run pylint checking only for docstrings `pylint --disable=all --enable=missing-docstring test_file.py`
4. Remove the underscore(s) from the method
5. Run pylint as in step 3. An error should now appear.

### Current behavior
No error is reported for the missing docstring in the private methods.

### Expected behavior
An error should be reported for every missing docstring.

### pylint --version output
```bash
$ pylint --version
No config file found, using default configuration
pylint 1.8.2, 
astroid 1.6.1
Python 3.6.3 (default, Oct  3 2017, 21:45:48) 
[GCC 7.2.0]
```

**Comment by brycepg** [comment]
This is actually intentional as per [PEP 8](https://www.python.org/dev/peps/pep-0008/#documentation-strings):

> Write docstrings for all public modules, functions, classes, and methods. Docstrings are not necessary for non-public methods, but you should have a comment that describes what the method does. This comment should appear after the def line.

**Comment by joshuahaertel** [comment]
Wow, I would have never noticed that. Sounds good!
